packages/lib-user/src/hooks/usePanoptesAuthToken.js

```diff
@@ -0,0 +1,35 @@
+import auth from "panoptes-client/lib/auth"
+import { useState } from "react"
+
+const isBrowser = typeof window !== "undefined"
+
+let defaultToken
+/*
+  Top-level await in modules has been supported in Node
+  and in all browsers since 2021. However, ES modules are still
+  not supported in the monorepo. An immediately-invoked async
```

Comment:
Perhaps this is picky, but I think "ES Modules are still not supported in the monorepo" is a misleading phrase that's been thrown around over the past two years. Yes ,ESM are incompatible with our test suites, see https://github.com/zooniverse/front-end-monorepo/issues/3779, but not a problem for FEM apps and libraries in a browser.

Reply:
None of the monorepo packages are ES modules. They all use `type: commonjs` in their package files by default, even if they have ES6 exports. Top level await is only available in modules, so I couldn’t use it here. 

If I could use top level await, the code would simplify to this, which would be great. 

```js
await auth.checkCurrent()
const defaultToken = await auth.checkBearerToken()
```

Reply:
By the way, Node 23 is out now and finally fixes static imports for CJS.

Reply:
> However, server-side rendering of ES modules in Node would require either type: module in package files, or .mjs extensions on modules.

Doesn't app-root do this?

Reply:
The changes here are in `lib-user`, which uses CommonJS in Node.

Reply:
You're right about `app-root`. Top level await will work in that package.

Reply:
Here's the build warning that I get from Next.js when I use top level await here.

```sh
printf 'Building `fe-root`...\n'
Building `fe-root`...
yarn workspace @zooniverse/fe-root build
$ APP_ENV=${APP_ENV:-development} NODE_ENV=${NODE_ENV:-production} PANOPTES_ENV=${PANOPTES_ENV:-production} next build
  ▲ Next.js 14.2.15

   Creating an optimized production build ...
 ⚠ Compiled with warnings

../lib-user/dist/esm/hooks/usePanoptesAuthToken.js
The generated code contains 'async/await' because this module is using "topLevelAwait".
However, your target environment does not appear to support 'async/await'.
As a result, the code may not run as expected or may cause runtime errors.
```